### PR TITLE
added some help texts to the Security and Privacy profile settings

### PR DESCRIPTION
**Pull request by tobiasd**

### CODE REVIEW

**Comment by annando** [comment]
I will leave the spellcheck to @MrPetovan :-)

File: mod/settings.php [hunk 1]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
 		$profile_in_dir = '<input type="hidden" name="profile_in_directory" value="1" />';
 	} else {
 		$profile_in_dir = replace_macros($opt_tpl, [
-			'$field' => ['profile_in_directory', L10n::t('Publish your default profile in your local site directory?'), $profile['publish'], L10n::t("Your profile may be visible in public."), [L10n::t('No'), L10n::t('Yes')]]
+			'$field' => ['profile_in_directory', L10n::t('Publish your default profile in your local site directory?'), $profile['publish'], L10n::t('If you enable this option your profile will be published in the global friendica directories (e.g. <a href="%s">%s</a>). Your profile may be visible in public.', Config::get('system', 'directory'), Config::get('system', 'directory')), [L10n::t('No'), L10n::t('Yes')]]
```
Comment (MrPetovan) [inline mod/settings.php]: `L10n::t('Your profile will be published in the global friendica directories (e.g. <a href="%1%s">%1%s</a>). Your profile details will be publicly visible.', Config::get('system', 'directory'))`

File: mod/settings.php [hunk 2]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
...
 		]);
 	}
 
 	if (strlen(Config::get('system', 'directory'))) {
 		$profile_in_net_dir = replace_macros($opt_tpl, [
-			'$field' => ['profile_in_netdirectory', L10n::t('Publish your default profile in the global social directory?'), $profile['net-publish'], '', [L10n::t('No'), L10n::t('Yes')]]
+			'$field' => ['profile_in_netdirectory', L10n::t('Publish your default profile in the global social directory?'), $profile['net-publish'], L10n::t('If you enable this option your profile will be listed in this nodes <a href="%s">local directory</a>.', System::baseUrl().'/directory'), [L10n::t('No'), L10n::t('Yes')]]
```
Comment (MrPetovan) [inline mod/settings.php]: I believe you swapped the help texts between `profile_in_directory` and `profile_in_netdirectory`.

`'Your profile will be published in this node\'s <a href="%s">local directory</a>. Your profile details will be publicly visible.', System::baseUrl().'/directory'`

File: mod/settings.php [hunk 3]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
...
 		]);
 	} else {
 		$profile_in_net_dir = '';
 	}
 
 	$hide_friends = replace_macros($opt_tpl, [
-		'$field' => ['hide-friends', L10n::t('Hide your contact/friend list from viewers of your default profile?'), $profile['hide-friends'], '', [L10n::t('No'), L10n::t('Yes')]],
+		'$field' => ['hide-friends', L10n::t('Hide your contact/friend list from viewers of your default profile?'), $profile['hide-friends'], L10n::t('If you enable this option your contacts wont be shown on your profile page. If you add further profiles, you can decide separately for each profile to show your contact list there.'), [L10n::t('No'), L10n::t('Yes')]],
```
Comment (MrPetovan) [inline mod/settings.php]: `'Your contact list won\'t be shown on your default profile page. You can decide to show your contact list separately for each additional profile you create.'`

File: mod/settings.php [hunk 4]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
...
 	]);
 
 	$hide_wall = replace_macros($opt_tpl, [
-		'$field' => ['hidewall', L10n::t('Hide your profile details from unknown viewers?'), $a->user['hidewall'], L10n::t("If enabled, posting public messages to Diaspora and other networks isn't possible."), [L10n::t('No'), L10n::t('Yes')]],
+		'$field' => ['hidewall', L10n::t('Hide your profile details from unknown viewers?'), $a->user['hidewall'], L10n::t("If you enable this option unknown visitors of your profile page will only see your profile picture, your display name and the nick name you are using. Please note that if this is enabled posting public messages to Diaspora and other networks isn't possible."), [L10n::t('No'), L10n::t('Yes')]],
```
Comment (MrPetovan) [inline mod/settings.php]: Who are `unknown viewers`? Anonymous people? Anyone not registered on this server? Anyone not a direct contact? Anyone not followed? Anyone not following?

File: mod/settings.php [hunk 5]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
...
 	]);
 
 	$blockwall = replace_macros($opt_tpl, [
-		'$field' => ['blockwall', L10n::t('Allow friends to post to your profile page?'), (intval($a->user['blockwall']) ? '0' : '1'), '', [L10n::t('No'), L10n::t('Yes')]],
+		'$field' => ['blockwall', L10n::t('Allow friends to post to your profile page?'), (intval($a->user['blockwall']) ? '0' : '1'), L10n::t('If you enable this option your contacts may write new postings on your profile wall, if they are authentificated. These postings will be distributed to your other contacts.'), [L10n::t('No'), L10n::t('Yes')]],
```
Comment (MrPetovan) [inline mod/settings.php]: `'Your contacts may write posts on your profile wall. These posts will be distributed to your other contacts.'`

Are the posts global public or just server local? Do contacts need to be followers, followed?

File: mod/settings.php [hunk 6]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
...
 	]);
 
 	$blocktags = replace_macros($opt_tpl, [
-		'$field' => ['blocktags', L10n::t('Allow friends to tag your posts?'), (intval($a->user['blocktags']) ? '0' : '1'), '', [L10n::t('No'), L10n::t('Yes')]],
+		'$field' => ['blocktags', L10n::t('Allow friends to tag your posts?'), (intval($a->user['blocktags']) ? '0' : '1'), L10n::t('If you enable this option your contacts are allowed to tag your postings with additional tags.'), [L10n::t('No'), L10n::t('Yes')]],
```
Comment (MrPetovan) [inline mod/settings.php]: `'Your contacts can add additional tags to your posts.'`

Followers? Followed?

File: mod/settings.php [hunk 7]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
...
 	]);
 
 	$suggestme = replace_macros($opt_tpl, [
-		'$field' => ['suggestme', L10n::t('Allow us to suggest you as a potential friend to new members?'), $suggestme, '', [L10n::t('No'), L10n::t('Yes')]],
+		'$field' => ['suggestme', L10n::t('Allow us to suggest you as a potential friend to new members?'), $suggestme, L10n::t('If you like, Friendica will suggest your profile to new members of the network as potential interesting contact.'), [L10n::t('No'), L10n::t('Yes')]],
```
Comment (MrPetovan) [inline mod/settings.php]: `'Friendica may suggest new members to add you as a contact.'`

File: mod/settings.php [hunk 8]
```diff
@@ -1106,40 +1106,40 @@ function settings_content(App $a)
...
 	]);
 
 	$unkmail = replace_macros($opt_tpl, [
-		'$field' => ['unkmail', L10n::t('Permit unknown people to send you private mail?'), $unkmail, '', [L10n::t('No'), L10n::t('Yes')]],
+		'$field' => ['unkmail', L10n::t('Permit unknown people to send you private mail?'), $unkmail, L10n::t('If this option is enabled verified user from the Friendica network may send you private messages, even if they are not in your contact list.'), [L10n::t('No'), L10n::t('Yes')]],
```
Comment (MrPetovan) [inline mod/settings.php]: `'Friendica network users may send you private messages even if they are not in your contact list.'`

**Comment by tobiasd** [comment]
> I will leave the spellcheck to @MrPetovan :-)

Yes please! That text was written after a very short night and a potential burglar in the cellar (which was only a neighbour who needed to visit it at 4 in the morning and make some noise).

**Comment by tobiasd** [comment]
I wish that night would have been laughable. Sometimes emoticons are just not the right reaction.

File: mod/settings.php (hunk 1 above)
Comment (tobiasd) [reply mod/settings.php]: I tried using only one  parameter for the replacement, but the magic does not work.

Comment (tobiasd) [reply mod/settings.php]: Funny though, it does not work either. `sprintf` demands more arguments.

File: mod/settings.php (hunk 2 above)
Comment (tobiasd) [reply mod/settings.php]: > I believe you swapped the help texts between profile_in_directory and profile_in_netdirectory.

Why? Then they are in the wrong order for a long time, as I have only added the missing help text.

Comment (tobiasd) [reply mod/settings.php]: The details still may be visible for the public, not will be, depending on how the admin settings of the node are. Public meaning unknown visitors from the internet.

File: mod/settings.php (hunk 4 above)
Comment (tobiasd) [reply mod/settings.php]: Unknow visitors are all visitors not arriving over the magic-auth link.

File: mod/settings.php (hunk 5 above)
Comment (tobiasd) [reply mod/settings.php]: Is it really "other contacts"? The contact writing to the wall will get the posting as well.

Comment (tobiasd) [reply mod/settings.php]: Contacts are contacts who are authed with magic-auth in this case. Visibility is applied from your default settings IIRC.

**Comment by MrPetovan** [comment]
I reacted to the way you said it, not how frightening I'm sure it had to be.

File: mod/settings.php (hunk 4 above)
Comment (MrPetovan) [reply mod/settings.php]: So, anonymous visitors then? I believe this is a more common way of designate them.

Comment (MrPetovan) [reply mod/settings.php]: `'Hide your profile details from anonymous visitors?'`

`'Anonymous visitors will only see your profile picture, your display name and the nickname you are using on your profile page. Disables posting public messages to Diaspora and other networks.'`

**Pull request merged**